src/lib.rs

```diff
@@ -159,14 +159,19 @@ pub unsafe extern "C" fn WrapperDecompressImageEx(
 ) -> i32 {
     match catch_unwind(|| {
         // For back-compat with C++ version we allow decompression of images with zeros in DQT tables
-        let mut enabled_features = EnabledFeatures::all();
-        enabled_features.reject_dqts_with_zeros = false;
 
         // C++ version has a bug where it uses 16 bit math in the SIMD path and 32 bit math in the scalar path
         // depending on the compiler options. If use_16bit_dc_estimate=true, the decompression uses a back-compat
         // mode that considers it. The caller should set use_16bit_dc_estimate to true only for images that were
         // compressed by C++ version with relevant compiler options.
-        enabled_features.use_16bit_dc_estimate = use_16bit_dc_estimate;
+
+        // this is a bit of a mess since for a while we were encoded a mix of 16 and 32 bit math
+        // (hence the two parameters in features).
+
+        let mut enabled_features = EnabledFeatures {
+            use_16bit_dc_estimate: use_16bit_dc_estimate,
+            ..EnabledFeatures::compat_lepton_vector_read()
```

Comment:
compat_lepton_vector_read has use_16bit_adv_predict is set to true. I think it's wrong, it should be set to true only by the new header parser logic. I think that you need to use here compat_lepton_scalar_read (probably the vector_read version can be eliminated)

Reply:
> compat_lepton_vector_read has use_16bit_adv_predict is set to true. I think it's wrong, it should be set to true only by the new header parser logic. I think that you need to use here compat_lepton_scalar_read (probably the vector_read version can be eliminated)

use_16bit_adv_predict should generally be true by default since this was the behavior until now.

Reply:
Oh, I see. You are right. But then I have an opposite question - why in WrapperCompressImage you are using compat_lepton_vector_write that have both use_16bit flags set to true? Should not they both be false?

Reply:
16 bit for both is actually the most compatible option, since these files are readable by both the C++ and the (new) Rust version, since it picks up the change from the header. For future compression, I think it makes the most sense to use the 16 bit option to true.